### PR TITLE
New content data model

### PR DESCRIPTION
**Pull request by andygnewman**
Objectives: 
- More closely align the next-api content model with the data that now exists in Next's ES cluster.
- Reduce the amount of data manipulation associated with specific n-card presentation
- Do so in a way that continues to support existing content queries until all can be moved on to new content model.

To Do
- Add fields
- [x] that are in ES that are not already in graphQL content (or are in graphQL under a different name)
- [x] to reflect better naming conventions for computed fields (eg. primaryThemeTag rather than primaryTheme)
- [x] computed fields that will be used by n-teaser (eg. genreTag, teaserTag)
- Add deprecated flag to fields that;
- [x] will be deprecated from ES
- [x] change the name of fields from what is in ES
- [x] names that don't reflect the content (eg. primaryTheme in favour of primaryThemeTag)
- Change `concept` model
- [x] rename as `tag` (tbc)
- [x] add field names as per ES
- [x] add deprecated flag to fields that change name 
- [x] add relativeUrl field, as well as url field to be full url
- [x] extract headshot out of attributes  (will need to ensure headshot in author tag attributes)
- Change `image` model
- [x] add fields as per ES
- [x] add deprecated flag to fields that change name 
- Deprecate `author` model
- [x] add deprecated notes to author model
- Testing
- [x] Add unit tests for all new fields that in some way manipulate the data from ES

### CODE REVIEW

File: server/graphql/types/content.js [hunk 1]
```diff
@@ -73,6 +78,39 @@ const author = new graphql.GraphQLObjectType({
 	})
 });
 
+const alternativeTitlesObject = new graphql.GraphQLObjectType({
+	name: 'AlternativeTitles',
+	fields: () => ({
+		promotionalTitle: {
+			type: graphql.GraphQLString
+		}
+	})
```
Comment (ironsidevsquincy) [inline server/graphql/types/content.js]: does this need to be an object? will there be more than one field? i.e. can `alternativeTitles` just be a `new graphql.GraphQLList(graphql.String)`

File: server/graphql/types/content.js [hunk 2]
```diff
@@ -73,6 +78,39 @@ const author = new graphql.GraphQLObjectType({
...
+});
+
+const commentsObject = new graphql.GraphQLObjectType({
+	name: 'Comments',
+	fields: () => ({
+		enabled: {
+			type: graphql.GraphQLBoolean
+		}
+	})
+});
```
Comment (ironsidevsquincy) [inline server/graphql/types/content.js]: guess it's possible more fields will be added, so 👍

File: test/server/graphql/types/tag.spec.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+	// xcontext('Fields that rely on further content calls', () => {
+	//
+	// 	xit('calls the search content backend to get latest content', () => {
+	// 		const query = 'query { tag { latestContent { id } } }';
+	// 		return getTagData(query, tagFixturePartial)
+	// 		.then((tag) => {
+	// 			expect(tag.latestContent).to.be.defined;
+	// 		});
+	// 	});
+	// });
```
Comment (ironsidevsquincy) [inline test/server/graphql/types/tag.spec.js]: delete?

File: test/server/graphql/types/content.spec.js [hunk 4]
```diff
@@ -67,6 +67,549 @@ describe('Content', () => {
...
+
+			it('should return the properties of main Image', () => {
+				const fixture = { mainImage: {
+														title: 'title',
+														url: 'url',
+														width: 900,
+														height: 600,
+														ratio: 1.5
+													}
+												};
```
Comment (andygout) [inline test/server/graphql/types/content.spec.js]: Funky indentation.

File: server/graphql/types/content.js (hunk 1 above)
Comment (andygnewman) [inline server/graphql/types/content.js]: Yes have left this open, but have also added the ability to pull out promotionalTitle on it's own,

File: test/server/graphql/types/tag.spec.js (hunk 3 above)
Comment (andygnewman) [inline test/server/graphql/types/tag.spec.js]: Yep, will do

**Comment by andygnewman** [comment]
Have made the amends requested and tested on front page and all was same as live version.

**Pull request merged**